SlackBotPrototype/API.Tests/EmbeddedResourceTestBase.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace API.Tests
+{
+	public class EmbeddedResourceTestBase

```

Comment:
Yes, both make sense in this case for the reason you specified as alternatives for code sharing. Some considerations

* In C# we can only inherit one class, so we're preventing ourselves from inheriting another class -- which might have state.
* By inheriting we're making an assuming that all future additions will be relevant for all inheritors. If we later added instance variables or methods that don't apply, we end with bloated object with unnecessary functionality.
* By inheriting we can limit the scope of our functions using `protected`, this limits access to our function more than a public utility method would. So caller's know that if they want to use this method, they have to inherit which means that the subclass has a relationship with the base class and it's functionality.
* If we used a public utility method, we're saying the caller can more freely use the method in different places without necessarily considering the relationship.